packages/playwright/src/matchers/toMatchText.ts

```diff
@@ -60,7 +60,13 @@ export async function toMatchText(
 
   const timeout = options.timeout ?? this.timeout;
 
-  const { matches: pass, received, log, timedOut } = await query(!!this.isNot, timeout);
+  const { matches: pass, received, log, timedOut } = await query(!!this.isNot, timeout).catch(async error => {
+    // FIXME: query should not throw, but it does for strict mode violations for example.
+    if (receiverType === 'Locator')
```

Comment:
Why only Locator and not Page?

Reply:
We only recover from errors on locators.